lib/jsonapi/plugs/query_parser.ex

```diff
@@ -111,7 +114,8 @@ defmodule JSONAPI.QueryParser do
     end
   end
 
-  def parse_fields(config, map) when map_size(map) == 0, do: config
+  @spec parse_fields(Config.t(), map()) :: Config.t() | no_return()
```

Comment:
Is no `no_return()` possible?

Reply:
Not on this particular definition of `parse_fields`, but the version on L120 it definitely is. Should I split up the typespec?

Reply:
Ah, here we go. If I split up the typspec Dialyzer...

```
lib/jsonapi/plugs/query_parser.ex:117:overlapping_contract
Overloaded contract for JSONAPI.QueryParser.parse_fields/2 has
overlapping domains; such contracts are currently unsupported and
are simply ignored.
```